src/main/java/codecain/GraphicalUserInterface/View/MenuGUI.java

```diff
@@ -10,58 +10,97 @@
+ */
 public class MenuGUI extends Application {
 
+    /**
+     * Initializes and displays the main menu window with options to launch the CLI or GUI.
+     *
+     * @param primaryStage The primary stage for this application.
+     */
     @Override
     public void start(Stage primaryStage) {
-        // Welcome message
         Text welcomeText = new Text("Welcome to Code Cain UML Editor");
         welcomeText.setFont(Font.font("Arial", 20));
 
-        // CLI Button
         Button cliButton = new Button("CLI");
         cliButton.setOnAction(event -> {
-            try {
-                Stage cliStage = new Stage();
-                CLI cli = new CLI();
-                cli.start(cliStage); // Launch CLI directly
-                primaryStage.hide(); // Hide the main menu
-
-                // Show the main menu again when the CLI stage is closed
-                cliStage.setOnCloseRequest(event1 -> primaryStage.show());
-            } catch (Exception e) {
-                e.printStackTrace();
-            }
+            launchCLI(primaryStage); 
         });
 
-        // GUI Button
         Button guiButton = new Button("Graphical Interface");
         guiButton.setOnAction(event -> {
-            try {
-                Stage guiStage = new Stage();
-                GraphicalInterfaceJavaFX gui = new GraphicalInterfaceJavaFX();
-                gui.start(guiStage); // Launch GUI directly
-                primaryStage.hide(); // Hide the main menu
-
-                // Show the main menu again when the GUI stage is closed
-                guiStage.setOnCloseRequest(event1 -> primaryStage.show());
-            } catch (Exception e) {
-                e.printStackTrace();
-            }
+            launchGraphicalInterface(primaryStage);
         });
-
-        // Layout setup
         VBox layout = new VBox(20, welcomeText, cliButton, guiButton);
-        layout.setAlignment(Pos.CENTER); // Center everything horizontally and vertically
-
-        // Scene setup
+        layout.setAlignment(Pos.CENTER);
         Scene scene = new Scene(layout, 400, 300);
         primaryStage.setTitle("UML Manager");
         primaryStage.setScene(scene);
         primaryStage.show();
     }
 
+    /**
+     * Launches the Command Line Interface (CLI) in a new stage.
+     * The main menu is hidden while the CLI window is active and shown again
+     * when the CLI window is closed.
+     *
+     * @param parentStage The main menu stage to hide while the CLI is active.
+     */
+    public static void launchCLI(Stage parentStage) {
+        try {
+            Stage cliStage = new Stage();
+            CLI cli = new CLI();
+            cli.start(cliStage);
+            if (parentStage != null) parentStage.hide(); // Hide the main menu
+
+            cliStage.setOnCloseRequest(event1 -> {
+                if (parentStage != null) parentStage.show();
+            });
+        } catch (Exception e) {
+            e.printStackTrace();
+        }
+    }
+
```

Comment:
I like how you created this helper method to call in the start method to simplify it. This is nice because then we are working on one stage.